### DISCUSSION TITLE
Does it work with Tensorflow?

### DISCUSSION THREAD

**Discussion by Block2122** [vosen/ZLUDA#66] [Q&A]
Hello everyone! Has anyone tried using this with Tensorflow 2? Do you have any opinions or advice to give me? Thanks in advance.

**Comment by DGdev91** [comment]
Isn't rocm supported on tensorflow already? what's the point?

**Reply by Block2122** [reply]
First of all, thanks for your answer! Maybe I'm missing something, but I have an RX 5700 XT and I can't get it to work with ROCm. Honestly, it's been a while since I checked to see if there have been any updates or improvements.

**Reply by DGdev91** [reply]
Well, the RX 5700xt is all another matter. I had the same gpu, and i had a long discussion on ROCm's github about it.
In short:
- 5700xt used to work on software compiled for rocm until 5.2 by setting "export HSA_OVERRIDE_GFX_VERSION=10.3.0". You can also use newer rocm as runtime, as long the software you are using has been compiled using the older version.
For Pytorch, for example, this means using the old pytorch 1.13 or an old nigthly build for pytorch 2.0
- On 5.3 and newer that workaround no longer works. Trying without the HSA Override doesn't work either, because some libraries are missing.
- 3 weeks ago, the issue has been been found. seems like that rocm tensile libs were not created when compiling wih the default setting, and because of that those are missing in current rocm packages. The PR has been fixed 2 weeks ago. https://github.com/ROCm/Tensile/pull/1862
- An amd developer also said he's planning to enable that on debian packages soon https://github.com/ROCm/ROCm/issues/2527

i guess it's just a matter of time and sooner or later all the major distros should have those libs packaged in

**Reply by insanie** [reply]
I can chip in by saying that it may be more convenient (not more effective) to translate CUDA in case with tensorflow, since building tensorflow for latest ROCm natively is a dependency hell and it only gets worse. Right now it requires more than 20 GB of dependencies and there's still instability.
Also, as Davide mentioned, allowed GPUs scope is getting narrower and HSA_OVERRIDE_GFX_VERSION override doesn't work anymore.

**Reply by UnisTorvalds** [reply]
[rocm_sdk_builder](https://github.com/lamikr/rocm_sdk_builder) is a neat tool which will simplify ROCm and TensorFlow installation for your gfx1010 (and many other AMD GPUs).
